sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala

```diff
@@ -43,6 +43,8 @@ class V2SessionCatalog(catalog: SessionCatalog)
   extends TableCatalog with FunctionCatalog with SupportsNamespaces with SQLConfHelper {
   import V2SessionCatalog._
 
+  var defaultColumnAnalyzer: Analyzer = null
```

Comment:
If possible, we should avoid using mutable variables.  
How about always using the `SimpleAnalyzer` to analyze the column default expression?  I can't recall why we pass in an analyzer in many column-default PRs, but `SimpleAnalyzer` seems enough for simple foldable expressions.

Reply:
This is my largest concern about this PR. We can have another PR for refactoring, whether before merging this one, or we can use `SimpleAnalyzer` in this PR and have a refactoring PR later.

Reply:
I like the idea of SimpleAnalyzerWithBuiltinFunctions, or we can make it ColumnDefaultSimpleAnalyzer.
It will be an associate Analyzer for the default column value constant folding. It is more reasonable than passing the Analyzer which contains the column default analysis rule to the column default analysis rule

Reply:
@gengliangwang SG, I started doing that change and it seemed better to create the refactoring first as a separate PR. I sent that out in https://github.com/apache/spark/pull/36880. Once that's merged, I can sync this PR with the latest changes and then it will be simpler.

Reply:
@gengliangwang Now that the refactoring PR is merged, I synced it into this PR and committed the change. This PR is now much simpler and ready for your review again, thank you for suggesting that ideas.